src/controllers/midi/legacymidicontrollermapping.cpp

```diff
@@ -20,8 +20,14 @@ void LegacyMidiControllerMapping::addInputMapping(uint16_t key, const MidiInputM
 }
 
 void LegacyMidiControllerMapping::removeInputMapping(uint16_t key) {
-    m_inputMappings.remove(key);
-    setDirty(true);
+    auto it = m_inputMappings.find(key);
+    if (it != m_inputMappings.end()) {
+        const MidiInputMapping& mapping = it.value();
+        if (!std::holds_alternative<std::shared_ptr<QJSValue>>(mapping.control)) {
+            setDirty(true);
+        }
+        m_inputMappings.erase(it);
+    }
```

Comment:
since `m_inputMappings` is a `QMultiHash` I assume multiple controls could be stored under the same key? `QMultiHash::remove` removes all mappings under that key while this only seems to remove the first one. I think you'll need something like:

```suggestion
    for (auto [it, end] = m_inputMappings.equal_range(key); it != end; ++it) {
        const MidiInputMapping& mapping = it.value();
        if (!std::holds_alternative<std::shared_ptr<QJSValue>>(mapping.control)) {
            setDirty(true);
        }
        m_inputMappings.erase(it);
    }
```

I'm not 100% sure how safe this is in regards to iterator invalidation because I wasn't able to find any docs on `QMultiHash::erase` (in Qt6 where `QMultiHash` doesn't publicly inherit from `QHash` anymore).